workbench-core/example/app/src/routes/datasetRoutes.ts

```diff
@@ -42,15 +43,50 @@ import {
   CreateExternalEndpoint,
   CreateExternalEndpointParser
 } from '../models/datasets/createExternalEndpoint';
+import { CreateImportDataSet, CreateImportDataSetParser } from '../models/datasets/createImportDataSet';
 import {
   CreatePresignedSinglePartFileUploadUrl,
   CreatePresignedSinglePartFileUploadUrlParser
 } from '../models/datasets/createPresignedFileUpload.ts';
+import { aws, dataSetsStoragePlugin } from '../services';
 import { wrapAsync } from '../utilities/errorHandlers';
 import { processValidatorResult } from '../utilities/validatorHelper';
 
 const timeToLiveSeconds: number = 60 * 1; // 1 minute
 
+async function getHostingAccountStoragePlugin(
+  roleToAssume: string,
+  externalId: string,
+  region: string
+): Promise<DataSetsStoragePlugin> {
+  try {
+    const { Credentials } = await aws.clients.sts.assumeRole({
+      RoleArn: roleToAssume,
+      RoleSessionName: 'Main-Account-Create-DataSet',
+      ExternalId: externalId
+    });
+
+    if (!Credentials) {
+      throw Boom.badRequest('Invalid roleToAssume param.');
+    }
+
+    const awsService = new AwsService({
+      region,
+      ddbTableName: process.env.DATASET_DDB_TABLE_NAME!,
```

Comment:
You're initializing the ddb connection to a table in the hosting account instead of the main account?

Reply:
Good call out. I'll remove the ddbTable from all hosting account awsServices

Reply:
fixed!